aios3/bucket.py

```diff
@@ -469,16 +468,26 @@ def __init__(self, bucket: Bucket):
         return response
 
     async def copy(self, CopySource, Key):
-        if isinstance(CopySource, dict):
-            path = amz_uriencode('/{}/{}'.format(CopySource['Bucket'], CopySource['Key']))
-            versionId = CopySource.get('VersionId')
-            if versionId:
-                path += '?versionId={}'.format(versionId)
-            CopySource = path
-        elif '?versionId=' not in CopySource:
-            CopySource = amz_uriencode(CopySource)
-
-        response = await self._request("PUT", '/' + Key, 'CopyObject', headers={'x-amz-copy-source': CopySource})
+        # transform to dictionary
+        if not isinstance(CopySource, dict):
+            src_bucket, src_key = CopySource.split('/', 1)
+            CopySource = {'Bucket': src_bucket, 'Key': src_key}
+
+            if '?versionId=' in src_key:
```

Comment:
What is the expected of the string version of CopySource? Is it a URL? This wouldn't be the normal way to parse a query string out of a URL